packages/zowe-explorer-ftp-extension/src/ZoweExplorerFtpUssApi.ts

```diff
@@ -81,6 +81,31 @@ export class FtpUssApi extends AbstractFtpApi implements ZoweExplorerApi.IUss {
         }
     }
 
+    /**
+     * Upload a file (located at the input path) to the destination path.
+     * @param inputFilePath The input file path
+     * @param ussFilePath The destination file path on USS
+     * @param options Any options for the upload
+     *
+     * @returns A file response containing the results of the operation.
+     */
+    public putContent(inputFilePath: string, ussFilePath: string, options?: zowe.IUploadOptions): Promise<zowe.IZosFilesResponse> {
+        return this.putContents(inputFilePath, ussFilePath, options?.binary, options?.localEncoding, options?.etag, options?.returnEtag);
+    }
+
+    /**
+     * Upload a file (located at the input path) to the destination path.
+     *
+     * @deprecated in favor of `putContent`
```

Comment:
should this be backwards? 
i.e. if `putContents` is deprecated in favor of `putContent`... shouldn't the `putContents` implementation be inside the `putContent` function,  and all `putContents` does is call `putContent` ?

---

Regarding: https://github.com/zowe/vscode-extension-for-zowe/pull/2082#discussion_r1123659534
> I guess the problem is that the IUss.putContents has all of the IUploadOptions as parameters instead of a single optional object.
What we could do it keep the inconsistency for V2 (i.e. IUss.putContents deprecated in favor of IUss.putContent) AND have a message/note in the IUss.putContent saying that it will be renamed in V3 😋

I was also wondering what we wanted to do about it. I'm in favor of keeping the plural forms, but I believe that `IUploadOptions vs all optional parameters` > `plural vs singular`. 😅 

If it's ok with you (and others), maybe we could leave a note in the `putContent` regarding the name change for V3? 😋

Reply:
we can't introduce breaking changes in 2.x.  I think FTP ext can either continue to use the old method or begin using the new as it is in code now.  v3 is where we can introduce the breaking changes.